s3pypi/package.py

```diff
@@ -133,12 +134,17 @@ def parse(html):
         return Index(Package(name, files) for name, files in filenames.items())
 
     def to_html(self):
+        for package in filter(lambda p: p in self.existing_files, self.packages):
+            package.files.update(self.existing_files[package])
         return self.template.render({"packages": self.packages})
 
     def add_package(self, package, force=False):
-        if force:
+        exists = list(filter(lambda p: p.version == package.version, self.packages))
+        if exists and force:
+            existing = exists[0]
+            self.existing_files[package] = existing.files
             self.packages.discard(package)
-        elif any(p.version == package.version for p in self.packages):
+        elif exists:
```

Comment:
Would it not be easier to just get the existing package and add its files to the new package? Then the `existing_files` dict is not needed, and `to_html` can remain unchanged. Something like this:
```python
existing = next((p for p in self.packages if p.version == package.version), None)
if existing and force:
    package.files.update(existing.files)
elif existing:
    raise S3PyPiError("...")
```
I only looked at this briefly, so let me know if I missed something.

Reply:
Sorry @natefoo, I edited my comment after the fact. I think adding the new files to the existing package should be fine (instead of the other way around).

Reply:
@natefoo I've tested a variation of this in 939b874024d9e674dfadccd0dc61b56936305318. If you want, you can merge those changes into your fork and then we'll merge this PR. Otherwise, I'll close this PR and open a new one.

Reply:
Ok, feel free to close this in favor of your solution.